src/classes/providers/base/types.ts

```diff
@@ -18,8 +18,8 @@ interface ICreateProxyConfig {
 interface IProviderConfig {
   host: string;
   port: number;
-  username: string;
-  password: string;
+  username?: string;
+  password?: string;
```

Comment:
not a fan of changing this interface. Should we handle and throw error for luminati and proxyrack that require these ?

Reply:
I've done it the other way around, and made proxyrack's & luminati's interface extends from a simpler IProviderConfig's one. This is up for debate, tell if if you'd rather see it the other way around.